File: Step 06 LinkedList/Step 6.1 1D Linked List/Insert Node at Beginning/Insert Node at Beginning.java


/*
class Node{
    int data;
    Node next;
    
    Node(int x){
        data = x;
        next = null;
    }
}
*/
class Solution
{
    //Function to insert a node at the beginning of the linked list.
    Node insertAtBeginning(Node head, int x)
    {
        // code here
        Node ptr = new Node(x);
        ptr.next = head;
        head = ptr;
        return head;
    }
    
    //Function to insert a node at the end of the linked list.
    Node insertAtEnd(Node head, int x)
    {
        // code here
        Node ptr = new Node(x);
        if(head == null){
            head = ptr;
            return head;
        }
        Node temp = head;
        while(temp.next!=null) temp = temp.next;
        temp.next = ptr;
        return head;
    }
}

File: Step 06 LinkedList/Step 6.1 1D Linked List/Insert Node at Beginning/Insert Node at Beginning.cpp
/*Structure of the linked list node is as
struct Node {
  int data;
  struct Node * next;
  Node(int x) {
    data = x;
    next = NULL;
  }
}; */

class Solution {
  public:
    Node *insertAtEnd(Node *head, int x) {
        // Code here
        Node* temp=head;
        if(head==NULL)
        {
            head=new Node(x);
            return head;
        }
        while(temp->next!=NULL)
        {
            temp=temp->next;
        }
        Node *temp2;
        temp2=new Node(x);
        temp->next=temp2;
        Node *pemp=head;
        return pemp;
    }
};